webapp/channels/src/components/post_edit_history/post_edit_history.tsx

```diff
@@ -37,25 +42,67 @@ const renderThumbVertical = (props: Record<string, unknown>): JSX.Element => (
 const PostEditHistory = ({
     channelDisplayName,
     originalPost,
-    postEditHistory,
 }: PropsFromRedux) => {
+    const [postEditHistory, setPostEditHistory] = useState<Post[]>([]);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [isLoading, setIsLoading] = useState<boolean>(false);
     const scrollbars = useRef<Scrollbars | null>(null);
     const {formatMessage} = useIntl();
+    const retrieveErrorHeading = formatMessage({
+        id: 'post_info.edit.history.retrieveError',
+        defaultMessage: 'Unable to load edit history',
+    });
+    const retrieveErrorSubheading = formatMessage({
+        id: 'post_info.edit.history.retrieveErrorVerbose',
+        defaultMessage: 'There was an error loading the history for this message. Check your network connection or try again later.',
+    });
 
     useEffect(() => {
+        const fetchPostEditHistory = async () => {
+            try {
+                setIsLoading(true);
+                const history = await Client4.getPostEditHistory(originalPost.id);
```

Comment:
Could this actually use the `getPostEditHistory` action instead of accessing the `Client4` instance directly? 

You'd have to change this to read the results object returned by that action to determine if there was an error instead of using a try/catch, but it should otherwise mostly be the same. We just want to limit the amount where components directly access the global `Client4` instance wherever we can

Reply:
Addressed in my most recent push - changed to leveraging `dispatch` and the `getPostEditHistory` redux action instead of the `Client4` function. Also adjusted how we perceive errors to be based on a lack of data rather than catching the error.

Reply:
Awesome, thanks